### DISCUSSION TITLE
What other register description formats have you used?

### DISCUSSION THREAD

**Discussion by amykyta3** [SystemRDL/systemrdl-compiler#122] [General]
Hi everyone!
I'm looking to expand support for new importers/exporters in the SystemRDL project toolchain, and wanted to survey what is popular in the industry. What other alternative input formats have you encountered?

I already have an [importer/exporter for IP-XACT](https://github.com/SystemRDL/PeakRDL-ipxact).
Just recently, I discovered the [CMSIS-SVD](https://www.keil.com/pack/doc/CMSIS/SVD/html/index.html) format and was hoping to build a similar translator.

Any other formats out there that are relatively popular?

**Comment by RasmusGOlsen** [comment]
I have mainly seen companies with their own proprietary format and generators. When I decides to migrate our proprietary format to SystemRDL I also looked at IP-XACT and the RALF format from Synopsys.

**Comment by xinwu-cyber** [comment]
As [Rasmus Olsen](https://github.com/RasmusGOlsen) mentioned,  RALF format from Synopsys would be required.

I developed two exporters, currently intended for internal testing:

- One to generate the Lauterbach TRACE32 register file (*.per).
- Another to generate the RestructuredText documentation (*.rst).
